app/qml/PanelHeader.qml

```diff
@@ -69,7 +69,7 @@ Rectangle {
             anchors.bottom: parent.bottom
             anchors.top: parent.top
             anchors.leftMargin: backIconVisible ? header.rowHeight/4 : 0
-            visible: ((title.contentWidth + backButton.width * 2) > header.width) || !backTextVisible ? false : true
+            visible: backTextVisible
```

Comment:
This is something that could look interesting. 
Since we use the text (namely "Cancel")  only for FeaturePanel, but we need the text there because the arrow icon is missing, I just get rid of previous condition which suppose to deal with long text.
However, _Cancel_ still might be a long text in some translations, which will probably end up with clash with a title.